prog/victoria_metrics/victoria_metrics_server_nexus.rb

```diff
@@ -0,0 +1,177 @@
+    vr = VictoriaMetricsResource[victoria_metrics_resource_id]
+    fail "No existing VictoriaMetricsResource" unless vr
+    Validation.validate_name(name)
+
+    DB.transaction do
+      ubid = VictoriaMetricsServer.generate_ubid
+      vm_st = Prog::Vm::Nexus.assemble_with_sshable(
+        Config.victoria_metrics_service_project_id,
+        sshable_unix_user: "ubi",
+        location_id: vr.location.id,
+        name: ubid.to_s,
+        size: vr.target_vm_size,
+        storage_volumes: [
+          {encrypted: true, size_gib: 30},
+          {encrypted: true, size_gib: vr.target_storage_size_gib}
+        ],
+        boot_image: "ubuntu-noble",
+        enable_ip4: true
+      )
+
+      vs = VictoriaMetricsServer.create(name: name, victoria_metrics_resource_id: victoria_metrics_resource_id, vm_id: vm_st.id) { _1.id = ubid.to_uuid }
+      Strand.create(prog: "VictoriaMetrics::VictoriaMetricsServerNexus", label: "start") { _1.id = vs.id }
+    end
+  end
+
+  def before_run
+    when_destroy_set? do
+      if strand.label != "destroy"
+        hop_destroy
+      elsif strand.stack.count > 1
+        pop "operation is cancelled due to the destruction of the VictoriaMetrics server"
+      end
+    end
+  end
+
+  label def start
+    nap 5 unless vm.strand.label == "wait"
+    victoria_metrics_server.incr_initial_provisioning
+
+    register_deadline("wait", 10 * 60)
+
+    cert, cert_key = create_certificate
+    victoria_metrics_server.update(cert: cert, cert_key: cert_key)
+
+    hop_bootstrap_rhizome
+  end
+
+  label def bootstrap_rhizome
+    bud Prog::BootstrapRhizome, {"target_folder" => "victoria_metrics", "subject_id" => vm.id, "user" => "ubi"}
+
+    hop_wait_bootstrap_rhizome
+  end
+
+  label def wait_bootstrap_rhizome
+    reap
+    hop_create_victoria_metrics_user if leaf?
+    donate
+  end
+
+  label def create_victoria_metrics_user
+    begin
+      victoria_metrics_server.vm.sshable.cmd("sudo groupadd -f --system victoria_metrics")
+      victoria_metrics_server.vm.sshable.cmd("sudo useradd --no-create-home --system -g victoria_metrics victoria_metrics")
+    rescue => ex
+      raise unless ex.message.include?("already exists")
+    end
+
+    hop_setup
+  end
+
+  label def setup
+    bud Prog::VictoriaMetrics::VictoriaMetricsSetup, {}, :install
+    bud Prog::VictoriaMetrics::VictoriaMetricsSetup, {}, :configure
+    bud Prog::VictoriaMetrics::VictoriaMetricsSetup, {}, :mount_data_disk
+
+    hop_wait_setup
+  end
+
+  label def wait_setup
+    reap
+    if leaf?
+      hop_wait
+    end
+    donate
+  end
+
+  label def wait
+    when_reconfigure_set? do
+      bud Prog::VictoriaMetrics::VictoriaMetricsSetup, {}, :configure
+      hop_wait_reconfigure
+    end
+
+    when_restart_set? do
+      decr_restart
+
+      push self.class, frame, "restart"
+    end
+
+    if victoria_metrics_server.certificate_last_checked_at < Time.now - 60 * 60 * 24 * 30 # ~1 month
+      hop_refresh_certificates
+    end
+
+    nap 10
```

Comment:
Similar to other prog, this can nap for long time without loss of functionality.